apps/web/src/modules/transaction/pages/detail/hooks/useTransactionDetail/index.ts

```diff
@@ -0,0 +1,45 @@
+import { GetPendingActorDebitsByJobDto } from '@modela/dtos'
+import { useErrorHandler } from 'common/hooks/useErrorHandler'
+import { apiClient } from 'common/utils/api'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+export const useTransactionDetail = () => {
+  const [transactionDetail, setTransactionDetail] =
+    useState<GetPendingActorDebitsByJobDto | null>()
+  const { handleError } = useErrorHandler()
+  const router = useRouter()
+  const { jobId } = router.query
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const res = await apiClient.get<GetPendingActorDebitsByJobDto>(
+          '/debits/jobs/' + jobId + '/actors',
+        )
+        setTransactionDetail(res.data)
+      } catch (err) {
+        handleError(err)
+      }
+    }
+    if (router.isReady) fetchData()
+  }, [handleError, jobId, router.isReady])
+
+  const actorFilter = (actorId: number) => {
+    if (!transactionDetail) return
+    let actorList = transactionDetail.actorList
+    actorList = actorList.filter((actor) => actor.actorId !== actorId)
+    setTransactionDetail({ ...transactionDetail, actorList })
```

Comment:
```suggestion
    setTransactionDetail((prevTransactionDetail) => ({ ...prevTransactionDetail, actorList }))
```

Reply:
Not sure why, but got type error when change this

Reply:
Because transactionList is either GetPendingActorDebitsByJobDto or null so it breaks spread operator if transactionList is null